radix-engine/src/updates/bottlenose.rs

```diff
@@ -1,557 +1,209 @@
-        (
-            scrypto_encode(&code_substate).unwrap(),
-            scrypto_encode(&vm_type_substate).unwrap(),
-            code_hash,
-        )
-    };
-
-    // Generate the new schema substate
-    let (
-        new_schema_substate,
-        get_current_time_input_v2_type_id,
-        compare_current_time_input_v2_type_id,
-        new_schema_hash,
-    ) = {
-        let mut aggregator = TypeAggregator::<ScryptoCustomTypeKind>::new();
-        let get_current_time_input_v2 =
-            aggregator.add_child_type_and_descendents::<ConsensusManagerGetCurrentTimeInputV2>();
-        let compare_current_time_input_v2 = aggregator
-            .add_child_type_and_descendents::<ConsensusManagerCompareCurrentTimeInputV2>();
-        let schema = generate_full_schema(aggregator);
-        let schema_hash = schema.generate_schema_hash();
-        let schema_substate = schema.into_locked_substate();
-        (
-            scrypto_encode(&schema_substate).unwrap(),
-            get_current_time_input_v2,
-            compare_current_time_input_v2,
-            schema_hash,
-        )
-    };
-
-    // Generate the blueprint definition substate updates
-    let updated_bp_definition_substate = {
-        let versioned_definition: VersionedPackageBlueprintVersionDefinition = reader
-            .read_object_collection_entry(
-                &consensus_mgr_pkg_node_id,
-                ObjectModuleId::Main,
-                ObjectCollectionKey::KeyValue(
-                    PackageCollection::BlueprintVersionDefinitionKeyValue.collection_index(),
-                    &bp_version_key,
-                ),
-            )
-            .unwrap()
-            .unwrap();
-
-        let mut definition = versioned_definition.fully_update_and_into_latest_version();
-
-        let export = definition
-            .function_exports
-            .get_mut(CONSENSUS_MANAGER_GET_CURRENT_TIME_IDENT)
-            .unwrap();
-        export.code_hash = code_hash;
-        let function_schema = definition
-            .interface
-            .functions
-            .get_mut(CONSENSUS_MANAGER_GET_CURRENT_TIME_IDENT)
-            .unwrap();
-        function_schema.input = BlueprintPayloadDef::Static(ScopedTypeId(
-            new_schema_hash,
-            get_current_time_input_v2_type_id,
-        ));
-
-        let export = definition
-            .function_exports
-            .get_mut(CONSENSUS_MANAGER_COMPARE_CURRENT_TIME_IDENT)
-            .unwrap();
-        export.code_hash = code_hash;
-        let function_schema = definition
-            .interface
-            .functions
-            .get_mut(CONSENSUS_MANAGER_COMPARE_CURRENT_TIME_IDENT)
-            .unwrap();
-        function_schema.input = BlueprintPayloadDef::Static(ScopedTypeId(
-            new_schema_hash,
-            compare_current_time_input_v2_type_id,
-        ));
+    /// Adds differed reference cost checks.    
+    pub ref_cost_checks: UpdateSetting<NoSettings>,
```

Comment:
Minor: can we rename it to `apply_costing_for_ref_checks`?

Reply:
Yup, I've renamed it